### PR TITLE
Move Python Console settings for font colors and family to new "Code Editor" tab in options dialog

### PR DESCRIPTION
**Pull request by nyalldawson**
These settings apply to more than just the Python console editors - rather they apply to all code editors across QGIS (including expression editors)

image

(ignore the fact that everything appears disabled in the screenshot -- that's a weird fedora 33 bug!!!)

### CODE REVIEW

File: src/app/options/qgscodeeditoroptions.cpp [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  // font
+  const QFont font = QgsCodeEditor::getMonospaceFont();
+  mFontComboBox->setCurrentFont( font );
+  mSizeSpin->setValue( font.pointSize() );
+  mOverrideFontGroupBox->setChecked( !settings.value( QStringLiteral( "codeEditor/fontfamily" ), QString(), QgsSettings::Gui ).toString().isEmpty() );
+}
+
+QString QgsCodeEditorOptionsWidget::helpKey() const
+{
+  return QStringLiteral( "plugins/python_console.html" );
```
Comment (nirvn) [inline src/app/options/qgscodeeditoroptions.cpp:124]: Should we remove the help until there's a specific doc covering this generic editor settings panel?

File: tests/src/python/test_qgscodeeditor.py [hunk 2]
```diff
@@ -61,6 +64,23 @@ def testColors(self):
         QgsCodeEditor.setColor(QgsCodeEditor.ColorRole.Keyword, QColor('#cc11bb'))
         self.assertEqual(QgsCodeEditor.color(QgsCodeEditor.ColorRole.Keyword).name(), '#cc11bb')
 
+    def testFontFamily(self):
+        f = QgsCodeEditor().getMonospaceFont()
+        self.assertTrue(f.styleHint() & QFont.Monospace)
```
Comment (nirvn) [inline tests/src/python/test_qgscodeeditor.py:69]: Nice to get some monospace font testing.

**Review by nirvn — APPROVED** [review]

File: src/app/options/qgscodeeditoroptions.cpp (hunk 1 above)
Comment (DelazJ) [reply src/app/options/qgscodeeditoroptions.cpp:124]:
```suggestion
  return QStringLiteral( "introduction/qgis_configuration.html#code-editor-settings" );
```
See https://github.com/qgis/QGIS-Documentation/pull/6169 (happy to get review)

**Comment by slarosa** [comment]
👏 👏 👏  excellent ...

**Comment by rduivenvoorde** [comment]
Super nice!!!!!!!!!!!!!!!!!!!!!
Also that you did it for all languages! 

What about giving the color texts some air on the left side? To me (also in the old situation) I intuitively ascociate the text most near to the color badge to be the name of the badge. Lemme try...

**Comment by nyalldawson** [comment]
@rduivenvoorde actually I'd suggest holding off on this -- I think @nirvn has plans for reorganising these buttons!

**Comment by rduivenvoorde** [comment]
Mmm, horizontal rulers are not behaving as I hoped:

Screenshot-20201005085913-712x327

**Comment by rduivenvoorde** [comment]
@nyalldawson ah, ok, thanks... was too eager to check :-)

**Pull request merged**